samples/android/Continote/app/src/main/java/com/firebasecontinue/sample/continote/EditNoteActivity.java

```diff
@@ -100,44 +173,180 @@ protected void handleUserSignedOut() {
     }
 
     /**
+     * Handles when the Note the user wants to edit has been found.
+     *
+     * @param note The Note that the user wishes to edit.
+     */
+    private void handleNoteToEditFound(Note note) {
+        // The Note to edit was found, so show and enable only the appropriate UI elements.
+        resetUiToInitialState();
+        setNoteEditorInputsToUseValuesFrom(note);
+        mNoteTitleTextInput.setEnabled(true);
+        mNoteContentTextInput.setEnabled(true);
+        mSaveButton.setEnabled(true);
+        mContinueWritingElsewhereButton.setEnabled(true);
+        mNoteFoundUiContainer.setVisibility(View.VISIBLE);
+    }
+
+    /**
+     * Handles when the Note the user wants to edit could not be found.
+     */
+    private void handleNoteToEditNotFound() {
+        // The Note to edit was not found, so show and enable only the appropriate UI elements.
+        resetUiToInitialState();
+        mNoteNotFoundUiContainer.setVisibility(View.VISIBLE);
+    }
+
+    /**
+     * Resets all UI elements (such as the Note editor inputs) to their initial state.
+     *
+     * The initial state is that no attempt has yet been made to get the Note from the Firebase
+     * Realtime Database, so neither the noteFoundUiContainer nor the noteNotFoundUiContainer
+     * elements will be shown or enabled.
+     */
+    private void resetUiToInitialState() {
+        mNoteFoundUiContainer.setVisibility(View.GONE);
+        mNoteNotFoundUiContainer.setVisibility(View.GONE);
+        mNoteTitleTextInput.setEnabled(false);
+        mNoteContentTextInput.setEnabled(false);
+        mSaveButton.setEnabled(false);
+        mContinueWritingElsewhereButton.setEnabled(false);
+        setNoteEditorInputsToUseValuesFrom(null);
+    }
+
+    /**
+     * Sets the Note editor's inputs to use the values from the provided Note.
+     *
+     * @param note The Note to use for the Note editor's inputs.
+     */
+    private void setNoteEditorInputsToUseValuesFrom(@Nullable Note note) {
+        mNoteTitleTextInput.setText((note != null) ? note.getTitle() : "");
+        mNoteContentTextInput.setText((note != null) ? note.getContent() : "");
+    }
+
+    /**
+     * Attempts to asynchronously set the value of the Note currently being edited in the Firebase
```

Comment:
Attempts implies there are reasonable and expected ways or reasons why this would fail. Is that the case? If so, document in the javadoc.